e2e/txpool_test.go

```diff
@@ -277,13 +273,28 @@ func TestTxPool_TransactionCoalescing(t *testing.T) {
 	// Add the transaction with the gap nonce value
 	addReq := generateReq(1)
 
-	_, addErr := clt.AddTxn(context.Background(), addReq)
+	addResp, addErr := clt.AddTxn(context.Background(), addReq)
 	if addErr != nil {
 		t.Fatalf("Unable to add txn, %v", addErr)
 	}
 
-	// Wait for the state transition to be executed
-	_ = waitForBlock(t, srv, 1, 0)
+	testTransactions = append(testTransactions, &testTransaction{
+		txHash: web3.HexToHash(addResp.TxHash),
+	})
+
+	for i := 1; i < len(testTransactions); i++ {
```

Comment:
You might want to add comments for an answer to the question "Why do you start from 1?"

Reply:
Added 👍 

[c2e5925](https://github.com/0xPolygon/polygon-edge/pull/485/commits/c2e592563352dfd8bec020d3954450c5ac0b32ff)